code/datums/outfit.dm

```diff
@@ -1,41 +1,153 @@
 	var/uniform = null
+
+	/// Type path of item to go in suit slot
 	var/suit = null
-	var/toggle_helmet = TRUE
+
+	/// Type path of item to go in back slot
 	var/back = null
+
+	/// Type path of item to go in belt slot
 	var/belt = null
+
+	/// Type path of item to go in gloves slot
 	var/gloves = null
+
+	/// Type path of item to go in shoes slot
 	var/shoes = null
+
+	/// Type path of item to go in head slot
 	var/head = null
+
+	/// Type path of item to go in mask slot
 	var/mask = null
+
+	/// Type path of item to go in neck slot
 	var/neck = null
+
+	/// Type path of item to go in ears slot
 	var/ears = null
+
+	/// Type path of item to go in the glasses slot
 	var/glasses = null
+
+	/// Type path of item to go in the idcard slot
 	var/id = null
+
+	/// Type path of item for left pocket slot
 	var/l_pocket = null
+
+	/// Type path of item for right pocket slot
 	var/r_pocket = null
+
+	/**
+	  * Type path of item to go in suit storage slot
+	  *
+	  * (make sure it's valid for that suit)
+	  */
 	var/suit_store = null
+
+	///Type path of item to go in the right hand
 	var/r_hand = null
+
+	//Type path of item to go in left hand
 	var/l_hand = null
-	var/internals_slot = null //ID of slot containing a gas tank
-	var/list/backpack_contents = null // In the list(path=count,otherpath=count) format
-	var/box // Internals box. Will be inserted at the start of backpack_contents
+
+	/// Should the toggle helmet proc be called on the helmet during equip
+	var/toggle_helmet = TRUE
+
+	///ID of the slot containing a gas tank
+	var/internals_slot = null
+
+	/**
+	  * list of items that should go in the backpack of the user
+	  *
+	  * Format of this list should be: list(path=count,otherpath=count)
+	  */
+	var/list/backpack_contents = null
+
+	/// Internals box. Will be inserted at the start of backpack_contents
+	var/box
+
+	/** 
+	  * Any implants the mob should start implanted with
+	  *
+	  * Format of this list is (typepath, typepath, typepath)
+	  */
 	var/list/implants = null
+
+	/// Any clothing accessory item
 	var/accessory = null
 
-	var/can_be_admin_equipped = TRUE // Set to FALSE if your outfit requires runtime parameters
-	var/list/chameleon_extras //extra types for chameleon outfit changes, mostly guns
+	/// Set to FALSE if your outfit requires runtime parameters
+	var/can_be_admin_equipped = TRUE
+
+	/**
+	  * extra types for chameleon outfit changes, mostly guns
+	  *
+	  * Format of this list is (typepath, typepath, typepath)
+	  *
+	  * These are all added and returns in the list for get_chamelon_diguise_info proc
+	  */
+	var/list/chameleon_extras
 
+/**
+  * Called at the start of the equip proc
+  *
+  * Override to change the value of the slots depending on client prefs, species and
+  * other such sources of change
+  *
+  * Extra Arguments
+  * * visualsOnly true if this is only for display (in the character setup screen)
+  *
+  * If visualsOnly is true, you can omit any work that doesn't visually appear on the character sprite
```

Comment:
should -> need , that's how you get weird bugs and runtimes.

Reply:
should does not appear in this line?

Reply:
do you mean can -> must?

Reply:
Yes, i was just hallucinating that one.